sdk/ai/azopenaiassistants/streaming_helpers_test.go

```diff
@@ -0,0 +1,237 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azopenaiassistants_test
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/ai/azopenaiassistants"
+	"github.com/stretchr/testify/require"
+)
+
+type streamScenario string
+
+const (
+	streamScenarioThreadAndRun streamScenario = "threadAndRun"
+	streamScenarioRun          streamScenario = "run"
+	streamScenarioTool         streamScenario = "tool"
+)
+
+// processStream checks that all the streaming events arrive and work the way you'd expect.
+//
+//   - threadAlreadyCreated - if you started your stream from CreateRun() you won't see the "thread created"
+//     event because _you_ already created the thread. This only fires off in the all-in-one CreateThreadAndRun.
+func processStream(t *testing.T, azure bool, asstID string, scenario streamScenario, stream *azopenaiassistants.EventReader[azopenaiassistants.StreamEvent]) *azopenaiassistants.ThreadRun {
+	type threadStats struct {
+		MessageContent string
+		EventCounts    map[azopenaiassistants.AssistantStreamEvent]int
+	}
+
+	thread := threadStats{
+		EventCounts: map[azopenaiassistants.AssistantStreamEvent]int{},
+	}
+
+	t.Logf("(%s) Stream has started", scenario)
+
+	var toolOutputsRequiredRet *azopenaiassistants.ThreadRun
+
+	for {
+		event, err := stream.Read()
+
+		if errors.Is(err, io.EOF) {
+			t.Logf("(%s) Stream has ended normally", scenario)
+			break
+		}
+
+		require.NoError(t, err)
+
+		thread.EventCounts[event.Kind]++
+
+		switch event.Kind {
```

Comment:
Can't we type switch on `event.Event`?

Reply:
I do wish we could but you'd still have to switch, within each 'case *ThreadRun', for instance, to see what the actual event was (ie, was it the thread run being created, the thread being completed, in progress, etc...

So in case that you'd have to switch twice, instead of once.